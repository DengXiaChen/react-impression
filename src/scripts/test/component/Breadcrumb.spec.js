import React from 'react';
import { shallow } from 'enzyme';
import { Breadcrumb } from '../../components/impression';

const setup = () => {
    const props = {
        separator: 'arrow',
    };

    const wrapper = shallow(
        <Breadcrumb {...props}>
            <a href="javascript:void(0);">首页</a>
            <a href="javascript:void(0);">财务管理</a>
            <span>收入</span>
        </Breadcrumb>
    );

    return {
        props,
        wrapper,
    };
};

describe('Breadcrumb', () => {
    const { wrapper } = setup();

    it('render', () => {
        expect(wrapper.find('.breadcrumb').exists()).toBe(true);
    });

    it('breadcrumb item', () => {
        expect(wrapper.find('.breadcrumb-item')).toHaveLength(3);
        // expect(wrapper.find('.breadcrumb-item::before')).toBe('>');
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
