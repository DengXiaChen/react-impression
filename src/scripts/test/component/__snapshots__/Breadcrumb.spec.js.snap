// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ol
    className="breadcrumb breadcrumb-arrow"
>
    <li
        className="breadcrumb-item"
    >
        <a
            href="javascript:void(0);"
        >
            首页
        </a>
    </li>
    <li
        className="breadcrumb-item"
    >
        <a
            href="javascript:void(0);"
        >
            财务管理
        </a>
    </li>
    <li
        className="breadcrumb-item"
    >
        <span>
            收入
        </span>
    </li>
</ol>,
  "nodes": Array [
    <ol
      className="breadcrumb breadcrumb-arrow"
>
      <li
            className="breadcrumb-item"
      >
            <a
                  href="javascript:void(0);"
            >
                  首页
            </a>
      </li>
      <li
            className="breadcrumb-item"
      >
            <a
                  href="javascript:void(0);"
            >
                  财务管理
            </a>
      </li>
      <li
            className="breadcrumb-item"
      >
            <span>
                  收入
            </span>
      </li>
</ol>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Breadcrumb
        separator="arrow"
>
        <a
                href="javascript:void(0);"
        >
                首页
        </a>
        <a
                href="javascript:void(0);"
        >
                财务管理
        </a>
        <span>
                收入
        </span>
</Breadcrumb>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": Array [
            <a
              href="javascript:void(0);"
>
              首页
</a>,
            <a
              href="javascript:void(0);"
>
              财务管理
</a>,
            <span>
              收入
</span>,
          ],
          "separator": "arrow",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ol
          className="breadcrumb breadcrumb-arrow"
>
          <li
                    className="breadcrumb-item"
          >
                    <a
                              href="javascript:void(0);"
                    >
                              首页
                    </a>
          </li>
          <li
                    className="breadcrumb-item"
          >
                    <a
                              href="javascript:void(0);"
                    >
                              财务管理
                    </a>
          </li>
          <li
                    className="breadcrumb-item"
          >
                    <span>
                              收入
                    </span>
          </li>
</ol>,
        "_debugID": 2,
        "_renderedOutput": <ol
          className="breadcrumb breadcrumb-arrow"
>
          <li
                    className="breadcrumb-item"
          >
                    <a
                              href="javascript:void(0);"
                    >
                              首页
                    </a>
          </li>
          <li
                    className="breadcrumb-item"
          >
                    <a
                              href="javascript:void(0);"
                    >
                              财务管理
                    </a>
          </li>
          <li
                    className="breadcrumb-item"
          >
                    <span>
                              收入
                    </span>
          </li>
</ol>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Breadcrumb
    separator="arrow"
>
    <a
        href="javascript:void(0);"
    >
        首页
    </a>
    <a
        href="javascript:void(0);"
    >
        财务管理
    </a>
    <span>
        收入
    </span>
</Breadcrumb>,
}
`;
